pipelines.py

```diff
@@ -44,12 +41,75 @@ def unet(config, train_mode):
 def unet_weighted(config, train_mode):
     unet_weighted = unet(config, train_mode)
     unet_weighted.get_step("loader").transformer = loaders.MetadataImageSegmentationLoaderDistances(**config.loader)
-    unet_weighted.get_step("unet").transformer = PyTorchUNetWeightedStream(
-        **config.unet) if config.execution.stream_mode else PyTorchUNetWeighted(
-        **config.unet)
+    unet_weighted.get_step("unet").transformer = PyTorchUNetWeightedStream(**config.unet) \
+        if config.execution.stream_mode else PyTorchUNetWeighted(**config.unet)
     return unet_weighted
 
 
+def unet_padded(config, train_mode):
+    if train_mode:
+        save_output = False
+        load_saved_output = False
+    else:
+        save_output = False
+        load_saved_output = False
+
+    if train_mode:
+        loader = _preprocessing_single_generator(config, train_mode, False)
+    else:
+        loader = _preprocessing_single_padded_generator(config)
+
+    unet = Step(name='unet',
+                transformer=PyTorchUNetStream(**config.unet) if config.execution.stream_mode else PyTorchUNet(
+                    **config.unet),
+                input_steps=[loader],
+                cache_dirpath=config.env.cache_dirpath,
+                save_output=save_output, load_saved_output=load_saved_output)
+
+    if train_mode:
+        mask_postprocessed = mask_postprocessing(loader, unet, config, save_output=save_output)
+    else:
+        prediction_crop = Step(name='prediction_crop',
+                               transformer=post.PredictionCropStream(
+                                   **config.postprocessor.prediction_crop) if config.execution.stream_mode \
+                                   else post.PredictionCrop(**config.postprocessor.prediction_crop),
+                               input_steps=[unet],
+                               adapter={'images': ([(unet.name, 'multichannel_map_prediction')]), },
+                               cache_dirpath=config.env.cache_dirpath,
+                               save_output=save_output)
+
+        prediction_renamed = Step(name='prediction_renamed',
+                                  transformer=Dummy(),
+                                  input_steps=[prediction_crop],
+                                  adapter={
+                                      'multichannel_map_prediction': ([(prediction_crop.name, 'cropped_images')]), },
+                                  cache_dirpath=config.env.cache_dirpath,
+                                  save_output=save_output)
+        mask_postprocessed = mask_postprocessing(loader, prediction_renamed, config, save_output=save_output)
```

Comment:
@jakubczakon  I was wondering if it wouldn't be better to pass an image adapter to mask_postprocessed , so that mask_postprocessed doesn't have hardcoded dependency on previous step and is more stand alone. That way there would be no need to use dummy step to only apply adapter.

Reply:
@taraspiotr I like the idea but it would take way more changes to get there.
We are passing loader too (with the output key implicitly assumed).
Now that I think about it our if our contract with mask_postprocessing is that the first positional has output key datagen and the second positional argument has output key multichannel_map_prediction it will all be good. So probably adding asserts on those is good enough. What do you think?

Also I think that sort of ad-hoc output name changes with somehting like rename_output='multichannel_map_prediction' that you can define in the step would be really cool. Especially since a lot of those Noop (Dummy) transformer shenanigans is exactly that.

Reply:
@jakubczakon I agree on both.